src/lib/sw_engine/tvgSwRaster.cpp

```diff
@@ -573,6 +573,66 @@ static bool _rasterImage(SwSurface* surface, uint32_t *img, uint32_t w, TVG_UNUS
     return true;
 }
 
+static uint32_t _applyBilinearInterpolation(const uint32_t *img, uint32_t w, uint32_t h, uint32_t rX, uint32_t rY, float fX, float fY)
+{
+    auto dX = fX - rX;
+    auto dY = fY - rY;
+
+    //0.0039f is a condition for the interpolation ratio to be at least 1.
+    if (abs(dX) > 0.0039f && abs(dY) > 0.0039f) {
```

Comment:
also because of this condition there are some small artifacts. I think that in case e.x. dX < 0.0039f we could still interpolate in the y dimension. what do you think?

Reply:
I deleted this if-check.
It seems rather unnecessary to check every pixel for this condition.